### PR TITLE
Add cwl-utils

### PR DESCRIPTION
**Pull request by kinow**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Closes https://github.com/common-workflow-language/cwl-utils/issues/138

### CODE REVIEW

**Comment by kinow** [comment]
Build failed due to missing `cwl-upgrader`. Will prepare the other recipe to unblock this PR.

**Comment by kinow** [comment]
@mr-c whenever you have time can you just reply here saying that you are OK being a maintainer of this recipe (pull request requirement). Thanks!

**Comment by mr-c** [comment]
I'm happy to maintain this package, yes!

**Comment by mr-c** [comment]
@kinow can you force push or otherwise add a commit to restart the build now that `cwl-upgrader` is packaged in conda?

**Comment by kinow** [comment]
> @kinow can you force push or otherwise add a commit to restart the build now that `cwl-upgrader` is packaged in conda?

Sure. Rebased and push-force'd. :drum:

**Comment by kinow** [comment]
@conda-forge/help-python , ready for review!

File: recipes/cwl-utils/meta.yaml
Comment (mr-c) [inline recipes/cwl-utils/meta.yaml:35]: You should run the tests as well

**Review by mr-c — CHANGES_REQUESTED** [review]

File: recipes/cwl-utils/meta.yaml
Comment (kinow) [reply recipes/cwl-utils/meta.yaml:35]: You mean calling each of `cwl-cite-extract`, `cwl-docker-extract`, `cwl-expression-refactor`, `cwl-graph-split`, `cwl-normalizer` to verify they were correctly added to the `$PATH`?

Comment (kinow) [reply recipes/cwl-utils/meta.yaml:41]: Added to the recipe the commands above.

File: recipes/cwl-utils/meta.yaml
Comment (kinow) [inline recipes/cwl-utils/meta.yaml:38]: Skipping this one for now as Conda failed to run it. Reproduced locally with `pip`. See https://github.com/common-workflow-language/cwl-utils/pull/161

File: recipes/cwl-utils/meta.yaml
Comment (mr-c) [reply recipes/cwl-utils/meta.yaml:41]: More about calling pytest post install https://github.com/conda-forge/cwltool-feedstock/blob/7640f7725c13c5b2c16be1c1037723dc5d9c761f/recipe/meta.yaml#L58

Comment (mr-c) [reply recipes/cwl-utils/meta.yaml:41]: Maybe add a `-k not ( tests_names requiring_cwlformat here )`

Comment (kinow) [reply recipes/cwl-utils/meta.yaml:41]: Ah, I see. Aren't we running the tests already in the `cwl-utils` repository? On Cylc we ran the tests only on the CI with GH Actions, and in Conda just the build for Conda and tests for imports & scripts only.

Comment (mr-c) [reply recipes/cwl-utils/meta.yaml:43]: Running the tests here can catch conda packaging specific issues

Comment (mr-c) [reply recipes/cwl-utils/meta.yaml:43]: `python -m pytest -n auto --pyargs cwl_utils -k "not test_graph_split"` 

FYI, this doesn't editable installs (`pip install -e .`)

**Comment by mr-c** [comment]
https://github.com/orgs/conda-forge/teams/help-python , cwl-utils is now no-arch and the entire test suite passes; a review and merge is appreciated to unblock updating the cwltool feedstock. Thanks! (the build fails on windows as schema-salad is not available there)

File: recipes/cwl-utils/meta.yaml
Comment (carterbox) [inline recipes/cwl-utils/meta.yaml:15]: List console script entry points in the build section.

https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

File: recipes/cwl-utils/meta.yaml
Comment (carterbox) [inline recipes/cwl-utils/meta.yaml:29]: These package specs do not match upstream.

https://github.com/common-workflow-language/cwl-utils/blob/v0.18/requirements.txt

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/cwl-utils/meta.yaml
Comment (kinow) [reply recipes/cwl-utils/meta.yaml]: Done. Now the only difference should be the python>3.6 and the cwlformat (it's an extra in the upstream https://github.com/common-workflow-language/cwl-utils/blob/v0.18/setup.py#L42).

File: recipes/cwl-utils/meta.yaml
Comment (kinow) [reply recipes/cwl-utils/meta.yaml:15]: Hmm, I couldn't find an example in the link above @carterbox. Had a look at the recently updated feedstock projects, but couldn't find an example of how to use the console script entry points in the build section. Would you have an example to share, please? Thanks.

Comment (carterbox) [reply recipes/cwl-utils/meta.yaml:15]: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points

Comment (kinow) [reply recipes/cwl-utils/meta.yaml:15]: Thanks @carterbox! I've updated the recipe, and also bumped to 0.19 (checked the requirements too)

**Comment by kinow** [comment]
Thank you @carterbox !

**Pull request merged**
